flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/ExpressionConverter.java

```diff
@@ -135,8 +138,26 @@ public RexNode visit(ValueLiteralExpression valueLiteral) {
 				return relBuilder.getRexBuilder().makeTimeLiteral(TimeString.fromCalendarFields(
 						valueAsCalendar(extractValue(valueLiteral, java.sql.Time.class))), 0);
 			case TIMESTAMP_WITHOUT_TIME_ZONE:
-				return relBuilder.getRexBuilder().makeTimestampLiteral(TimestampString.fromCalendarFields(
-						valueAsCalendar(extractValue(valueLiteral, java.sql.Timestamp.class))), 3);
+				TimestampType timestampType = (TimestampType) type;
+				Class<?> clazz = valueLiteral.getOutputDataType().getConversionClass();
```

Comment:
- First, these logical should in `extractValue`.
- Second, we should just use `valueLiteral.getValueAs`.

Reply:
We can just use `valueLiteral.getValueAs(LocalDateTime.class)`.